cargo-dist/src/tasks.rs

```diff
@@ -846,6 +846,22 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
     ) -> DistResult<Self> {
         let root_workspace_idx = workspaces.root_workspace_idx();
         let root_workspace = workspaces.workspace(root_workspace_idx);
+
+        // Complain if someone still has [workspace.metadata.dist] in a dist-workspace.toml scenario
+        if let Some(dist_manifest_path) = root_workspace.dist_manifest_path.as_deref() {
```

Comment:
are we turning this on now?

Reply:
This code only runs if someone has migrated to dist-workspace.toml, so this is basically just erroring on "it looks like you started a migration but I see some gunk you missed".